python_modules/dagster/dagster/_core/definitions/asset_graph_subset.py

```diff
@@ -42,7 +42,9 @@ def non_partitioned_asset_keys(self) -> AbstractSet[AssetKey]:
 
     @property
     def asset_keys(self) -> AbstractSet[AssetKey]:
-        return self.partitions_subsets_by_asset_key.keys() | self._non_partitioned_asset_keys
+        return {
```

Comment:
Fixes a bug I discovered -- a partitioned asset key should not be considered part of the subset if its partitions subset is empty

Reply:
good call